src/Core/Components/DateTime/FluentCalendarDay.cs

```diff
@@ -48,7 +48,7 @@ internal FluentCalendarDay(FluentCalendar calendar, DateTime day)
     /// <summary>
     /// Gets or sets a value indicating whether the day is selected by the user.
     /// </summary>
-    public bool IsSelected => Date.DayOfYear == _calendar.Value.GetValueOrDefault().DayOfYear;
+    public bool IsSelected => Date == _calendar.Value;
```

Comment:
Why did you remove this code? Sometime, the "date" are same but not the "time" part. You need to include these cases also.

Reply:
Oops, you are right. I missed the "time" part.
But the DayOfYear is not correct too!
It returns the nth value of the day in a year.
This way, the DayOfYear for both '2023-01-01' and '2024-01-01' are the same, however they are not the same date!
I'll fix it